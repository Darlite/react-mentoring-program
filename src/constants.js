export const genreNames = ['All', 'Documentary', 'Comedy', 'Horror', 'Crime'];