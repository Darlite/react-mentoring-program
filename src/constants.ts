export const genreNames: string[] = [
    'All',
    'Documentary',
    'Comedy',
    'Horror',
    'Crime'
];