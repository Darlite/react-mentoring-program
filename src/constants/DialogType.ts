export enum DialogType {
    AddMovie = "ADD MOVIE",
    EditMovie = "EDIT MOVIE",
    DeleteMovie = "DELETE MOVIE",
}
