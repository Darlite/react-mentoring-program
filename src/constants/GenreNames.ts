export const GenreNames: string[] = [
    'All',
    'Documentary',
    'Comedy',
    'Horror',
    'Crime'
];
