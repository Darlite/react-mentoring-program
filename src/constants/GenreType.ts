export enum GenreType {
    Action = "Action",
    Adventure = "Adventure",
    Comedy = "Comedy",
    Crime = "Crime",
    Documentary = "Documentary",
    Drama = "Drama",
    Fantasy = "Fantasy",
    Horror = "Horror",
    Musical = "Musical",
    Mystery = "Mystery",
    Romance = "Romance",
    ScienceFiction = "Science Fiction",
    Thriller = "Thriller",
    War = "War",
}