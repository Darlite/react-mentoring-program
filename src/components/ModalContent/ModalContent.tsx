import React from "react";
import {DialogType} from "../../constants/DialogType";
import MovieForm from "../MovieForm/MovieForm";
import {MovieDetailsData} from "../../types/MovieDetailsData";
import DeleteDialog from "../DeleteDialog/DeleteDialog";

interface ModalContentProps {
    currentDialog: string,
    selectedMovie: MovieDetailsData | null,
    handleSubmit: (MovieData: MovieDetailsData) => void,
}

const ModalContent: React.FC<ModalContentProps> = ({ currentDialog, selectedMovie, handleSubmit }) => {
        if (currentDialog === DialogType.EditMovie && selectedMovie) {
            return <MovieForm initialMovieInfo={selectedMovie} handleSubmit={handleSubmit} />;
        }

        if (currentDialog === DialogType.AddMovie) {
            return <MovieForm handleSubmit={handleSubmit} />
        }

        if (currentDialog === DialogType.DeleteMovie && selectedMovie) {
            return <DeleteDialog movieToDelete={selectedMovie} />;
        }

        return null
}

export default ModalContent;
