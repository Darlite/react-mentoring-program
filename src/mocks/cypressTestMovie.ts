import {GenreType} from "../constants/GenreType";

export const cypressTestMovie = {
    title: "Cypress Test Movie",
    editedMovieTitle: "Cypress Edited Test Movie",
    release_date: "2025-05-04",
    poster_path: "https://www.movieposters.com/cdn/shop/files/tron-ares_avxg7mih_480x.progressive.jpg",
    genres: [GenreType.Action, GenreType.ScienceFiction],
    runtime: "138",
    vote_average: "7.2",
    overview: "A highly sophisticated Program, Ares, is sent from the digital world into the real world on a dangerous mission.",
}
