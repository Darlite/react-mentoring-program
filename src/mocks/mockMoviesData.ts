import {GenreType} from "../constants/GenreType";

export const mockMoviesData = {
    id: 680,
    poster_path: "https://www.movieposters.com/cdn/shop/files/PulpFiction.Ver1_480x.progressive.jpg?v=1741192018",
    title: "Pulp Fiction",
    release_date: "1994-09-10",
    genres: [GenreType.Crime, GenreType.Thriller],
    runtime: "167",
    vote_average: "8.3",
    overview: "Jules Winnfield (Samuel L. Jackson) and Vincent Vega (John Travolta) are two hit men who are out to retrieve a suitcase stolen from their employer, mob boss Marsellus Wallace (Ving Rhames). Wallace has also asked Vincent to take his wife Mia (Uma Thurman) out a few days later when Wallace himself will be out of town. Butch Coolidge (Bruce Willis) is an aging boxer who is paid by Wallace to lose his fight. The lives of these seemingly unrelated people are woven together comprising of a series of funny, bizarre and uncalled-for incidents.â€”Soumitra",
}
