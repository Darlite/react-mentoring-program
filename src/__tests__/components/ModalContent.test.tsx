import ModalContent from "../../components/ModalContent/ModalContent";
import {render, screen} from "@testing-library/react";
import {DialogType} from "../../constants/DialogType";
import {mockMoviesData} from "../../mocks/mockMoviesData";

describe("ModalContent", () => {
    it("matches snapshot for Add Movie dialog", () => {
        const { container } = render(<ModalContent currentDialog={DialogType.AddMovie}
                                                   selectedMovie={mockMoviesData}
                                                   handleSubmit={jest.fn()}/>
        );

        expect(container).toMatchSnapshot();
    });

    it("matches snapshot for Edit Movie dialog", () => {
        const { container } = render(<ModalContent currentDialog={DialogType.EditMovie}
                                                   selectedMovie={mockMoviesData}
                                                   handleSubmit={jest.fn()}/>
        );

        expect(container).toMatchSnapshot();
    });

    it("matches snapshot for Delete Movie dialog", () => {
        const { container } = render(<ModalContent currentDialog={DialogType.DeleteMovie}
                                                   selectedMovie={mockMoviesData}
                                                   handleSubmit={jest.fn()}/>
        );

        expect(container).toMatchSnapshot();
    });

    it('renders MovieForm for Add Movie dialog', () => {
        render(
            <ModalContent currentDialog={DialogType.AddMovie}
                          handleSubmit={jest.fn()}
                          selectedMovie={mockMoviesData}
            />
        );

        expect(screen.getByPlaceholderText("Movie title")).toBeInTheDocument();
        expect(screen.queryByDisplayValue("Pulp Fiction")).not.toBeInTheDocument();
    });

    it('renders MovieForm for Edit Movie dialog', () => {
        render(
            <ModalContent currentDialog={DialogType.EditMovie}
                          selectedMovie={mockMoviesData}
                          handleSubmit={jest.fn()} />
        );

        expect(screen.getByPlaceholderText("Movie title")).toBeInTheDocument();
        expect(screen.getByDisplayValue(mockMoviesData.title)).toBeInTheDocument();
    });

    it('renders MovieForm for Delete Movie dialog', () => {
        render(
            <ModalContent currentDialog={DialogType.DeleteMovie}
                          selectedMovie={mockMoviesData}
                          handleSubmit={jest.fn()} />
        );

        expect(screen.getByText(/Are you sure you want to delete/)).toBeInTheDocument();
        expect(screen.getByRole("button", { name: "Confirm" })).toBeInTheDocument();
    });

    it('renders nothing if the proper DialogType is not provided', () => {
        const { container } = render(<ModalContent currentDialog={""}
                                                   selectedMovie={mockMoviesData}
                                                   handleSubmit={jest.fn()} />
        );

        expect(container).toBeEmptyDOMElement();
    })
})
