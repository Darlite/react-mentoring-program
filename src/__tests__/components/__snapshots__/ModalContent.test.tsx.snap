// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalContent matches snapshot for Add Movie dialog 1`] = `
<div>
  <form
    class="movieForm"
  >
    <div
      class="labelAndInputContainer"
    >
      <label
        class="inputLabel"
        for="movieTitle"
      >
        Title
      </label>
      <input
        aria-label="Movie title"
        class="input"
        id="movieTitle"
        name="title"
        placeholder="Movie title"
        required=""
        type="text"
        value=""
      />
    </div>
    <div
      class="labelAndInputContainer"
    >
      <label
        class="inputLabel"
        for="releaseDate"
      >
        Release Date
      </label>
      <input
        aria-label="Release Date"
        class="input"
        id="releaseDate"
        name="release_date"
        placeholder="Select Date"
        required=""
        type="date"
        value=""
      />
    </div>
    <div
      class="labelAndInputContainer"
    >
      <label
        class="inputLabel"
        for="movieUrl"
      >
        Movie URL
      </label>
      <input
        aria-label="Movie URL"
        class="input"
        id="movieUrl"
        name="movieUrl"
        placeholder="https://"
        required=""
        type="url"
        value=""
      />
    </div>
    <div
      class="labelAndInputContainer"
    >
      <label
        class="inputLabel"
        for="rating"
      >
        Rating
      </label>
      <input
        aria-label="Movie Rating"
        class="input"
        id="rating"
        name="vote_average"
        placeholder="7.8"
        required=""
        type="number"
        value=""
      />
    </div>
    <div
      class="labelAndInputContainer"
    >
      <label
        class="inputLabel"
        for="genres"
      >
        Genre
      </label>
      <select
        aria-label="Movie Genres"
        class="input"
        id="genres"
        multiple=""
        name="genres"
        required=""
      >
        <option
          value="Action"
        >
          Action
        </option>
        <option
          value="Crime"
        >
          Crime
        </option>
        <option
          value="Documentary"
        >
          Documentary
        </option>
        <option
          value="Thriller"
        >
          Thriller
        </option>
        <option
          value="Musical"
        >
          Musical
        </option>
      </select>
    </div>
    <div
      class="labelAndInputContainer"
    >
      <label
        class="inputLabel"
        for="runtime"
      >
        Runtime
      </label>
      <input
        aria-label="Movie Runtime"
        class="input"
        id="runtime"
        name="runtime"
        placeholder="minutes"
        required=""
        type="text"
        value=""
      />
    </div>
    <div
      class="overviewContainer"
    >
      <label
        class="inputLabel"
        for="overview"
      >
        Overview
      </label>
      <textarea
        aria-label="Movie overview"
        class="input"
        id="overview"
        name="overview"
        placeholder="Movie overview"
      />
    </div>
    <div
      class="buttonsContainer"
    >
      <button
        aria-label="Reset"
        class="resetButton"
        type="reset"
      >
        Reset
      </button>
      <button
        aria-label="Submit"
        class="submitButton"
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</div>
`;

exports[`ModalContent matches snapshot for Delete Movie dialog 1`] = `
<div>
  <p
    class="deleteMessage"
  >
    Are you sure you want to delete
    <span
      aria-label="movie title"
      class="movieToDeleteTitle"
    >
       
      Pulp Fiction
       
    </span>
    movie?
  </p>
  <button
    aria-label="Confirm"
    class="confirmDeleteMovieButton"
  >
    Confirm
  </button>
</div>
`;

exports[`ModalContent matches snapshot for Edit Movie dialog 1`] = `
<div>
  <form
    class="movieForm"
  >
    <div
      class="labelAndInputContainer"
    >
      <label
        class="inputLabel"
        for="movieTitle"
      >
        Title
      </label>
      <input
        aria-label="Movie title"
        class="input"
        id="movieTitle"
        name="title"
        placeholder="Movie title"
        required=""
        type="text"
        value="Pulp Fiction"
      />
    </div>
    <div
      class="labelAndInputContainer"
    >
      <label
        class="inputLabel"
        for="releaseDate"
      >
        Release Date
      </label>
      <input
        aria-label="Release Date"
        class="input"
        id="releaseDate"
        name="release_date"
        placeholder="Select Date"
        required=""
        type="date"
        value="1994-09-23"
      />
    </div>
    <div
      class="labelAndInputContainer"
    >
      <label
        class="inputLabel"
        for="movieUrl"
      >
        Movie URL
      </label>
      <input
        aria-label="Movie URL"
        class="input"
        id="movieUrl"
        name="movieUrl"
        placeholder="https://"
        required=""
        type="url"
        value="https://www.imdb.com/title/tt0110912"
      />
    </div>
    <div
      class="labelAndInputContainer"
    >
      <label
        class="inputLabel"
        for="rating"
      >
        Rating
      </label>
      <input
        aria-label="Movie Rating"
        class="input"
        id="rating"
        name="vote_average"
        placeholder="7.8"
        required=""
        type="number"
        value="8.9"
      />
    </div>
    <div
      class="labelAndInputContainer"
    >
      <label
        class="inputLabel"
        for="genres"
      >
        Genre
      </label>
      <select
        aria-label="Movie Genres"
        class="input"
        id="genres"
        multiple=""
        name="genres"
        required=""
      >
        <option
          value="Action"
        >
          Action
        </option>
        <option
          selected=""
          value="Crime"
        >
          Crime
        </option>
        <option
          value="Documentary"
        >
          Documentary
        </option>
        <option
          selected=""
          value="Thriller"
        >
          Thriller
        </option>
        <option
          value="Musical"
        >
          Musical
        </option>
      </select>
    </div>
    <div
      class="labelAndInputContainer"
    >
      <label
        class="inputLabel"
        for="runtime"
      >
        Runtime
      </label>
      <input
        aria-label="Movie Runtime"
        class="input"
        id="runtime"
        name="runtime"
        placeholder="minutes"
        required=""
        type="text"
        value="2h 34m"
      />
    </div>
    <div
      class="overviewContainer"
    >
      <label
        class="inputLabel"
        for="overview"
      >
        Overview
      </label>
      <textarea
        aria-label="Movie overview"
        class="input"
        id="overview"
        name="overview"
        placeholder="Movie overview"
      >
        Jules Winnfield (Samuel L. Jackson) and Vincent Vega (John Travolta) are two hit men who are out to retrieve a suitcase stolen from their employer, mob boss Marsellus Wallace (Ving Rhames). Wallace has also asked Vincent to take his wife Mia (Uma Thurman) out a few days later when Wallace himself will be out of town. Butch Coolidge (Bruce Willis) is an aging boxer who is paid by Wallace to lose his fight. The lives of these seemingly unrelated people are woven together comprising of a series of funny, bizarre and uncalled-for incidents.â€”Soumitra
      </textarea>
    </div>
    <div
      class="buttonsContainer"
    >
      <button
        aria-label="Reset"
        class="resetButton"
        type="reset"
      >
        Reset
      </button>
      <button
        aria-label="Submit"
        class="submitButton"
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</div>
`;
